cmd/capabilities/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,169 @@
+// Sample run-helloworld is a minimal Cloud Run service.
+package main
+
+import (
+        "log"
+        "time"
+        "os"
+        "context"
+        "runtime"
+        "encoding/json"
+        "cloud.google.com/go/pubsub"
+        "github.com/apigee/registry/rpc"
+	    "github.com/golang/protobuf/jsonpb"
+        "google.golang.org/grpc/codes"
+        "google.golang.org/grpc/status"
+        "github.com/apigee/registry/cmd/capabilities/utils"
+        cloudtasks "cloud.google.com/go/cloudtasks/apiv2"
+        taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2"
+)
+
+// Make this configurable
+const topicName = "changes"
+const subscriptionName = topicName + "-pull-subscriber"
+
+func main() {
+    log.Print("Starting subscriber...")
+    ctx := context.Background()
+
+    // Setup and start the dispatcher server
+    dispatcher := &Dispatcher{}
+    err := dispatcher.setUp(ctx)
+    if err != nil {
+        log.Printf(err.Error())
+        return
+    }
+    err = dispatcher.startServer(ctx)
+    if err != nil {
+        log.Printf(err.Error())
+    }
+    return
+}
+
+type Dispatcher struct {
```

Comment:
Do we intend on having this struct provide any functionality beyond setting up and starting a server? Requiring a call to `setUp` before its other methods indicates we're putting too much responsibility on the user of the type.

It seems more appropriate to create a package `dispatcher` with exported method `StartServer(context.Context)` that handles any necessary setup internally. The main package would become very lightweight with its only responsibility being starting the server (possibly parsing configuration info in the future).

Reply:
Planning to refactor in another PR. Tracking issue: https://github.com/apigee/registry/issues/100